### PR TITLE
User actions

### PR DESCRIPTION
**Pull request by DMCFA**
get user

### CODE REVIEW

File: src/middleware.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { authMiddleware } from '@clerk/nextjs';
+
+export default authMiddleware({
+  publicRoutes: ['/'],
+});
+
+export const config = {
+  matcher: ['/((?!.+\\.[\\w]+$|_next).*)', '/', '/(api|trpc)(.*)'],
```
Comment (AnthonyClarke25) [inline src/middleware.ts]: what's this doing? i see regex my brain goes blah

File: src/app/sign-up/[[...sign-up]]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { SignUp } from '@clerk/nextjs';
+
+export default function Page() {
+  return (
+    <div className='flex items-center justify-center min-h-screen'>
```
Comment (AnthonyClarke25) [inline src/app/sign-up/[[...sign-up]]/page.tsx]: these classes where do they come from? is this tailwind?

**Review by AnthonyClarke25 — APPROVED** [review]

File: src/middleware.ts (hunk 1 above)
Comment (DMCFA) [reply src/middleware.ts]: all routes are protected by Clerk's authentication middleware, with the exception of internal /_next/ routes and static files

File: src/app/sign-up/[[...sign-up]]/page.tsx (hunk 2 above)
Comment (DMCFA) [reply src/app/sign-up/[[...sign-up]]/page.tsx]: yes, the idea is to use tailwind in this project

**Pull request merged**